internal/collector/syslogprocessor/processor_test.go

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) F5, Inc.
+//
+// This source code is licensed under the Apache License, Version 2.0 license found in the
+// LICENSE file in the root directory of this source tree.
+
+package syslogprocessor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/leodido/go-syslog/v4/rfc3164"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/processor/processortest"
+	"go.uber.org/zap"
+)
+
+//nolint:lll,revive // long test string kept for readability
+func TestSyslogProcessor(t *testing.T) {
+	testCases := []struct {
+		expectAttrs   map[string]string
+		body          any
+		name          string
+		expectJSON    string
+		expectRecords int
+	}{
+		{
+			name: "csv nginx app protect syslog message",
```

Comment:
Just for the test case names can you use this kind of format in the naming like this:
`Test 1: CSV NGINX App Protect syslog message`